### DISCUSSION TITLE
Growatt SPH3600 not providing data to HA

### DISCUSSION THREAD

**Discussion by kl1127** [wills106/homeassistant-solax-modbus#514] [Q&A]
Hi @wills106 

My inverter, a Growatt SPH3600 has the serial number OMCUCC904T and "property" Property: RA1.0/RAAA212006/ZCBA-0006/A1B9D0T0PFU2M3S7

Might I ask if you could add this serial? I am getting no entities through the USR-W610 I have installed.

As I am not sure if the problem is getting data from the inverter to the USR-W610 or from the USR-W610 to HA, I am giving you a few more details. I have tried different pins.

Setup SolaX integration:
Interface "TCP / Ethernet"
Modbus address "1"
Inverter type "growatt"
Polling "15"
EPS ticked (as I am running in EPS only at the moment)
IP set
TCP port "502"
Modbus RTU over TCP

Setup on USR-W610:
Baud rate "9600"
Data bits "8"
Parity "none"
Stop "1"
Flow control "Disable"
485 mode "Enable"
Baud rate adaptive "Enable"
STA Mode, Modbus TCP <=> Modbus RTU

I am getting the following in the log in HA:
reading firmware number from address 0x9 failed; other address may succeed
attempt to read firmware failed at 0x9
attempt to read firmware failed at 0xbb9
reading firmware number from address 0xbb9 failed; other address may succeed
cannot find firmware version, even not for other Inverter
firmware version : unknown
Setup of sensor platform growatt_server is taking over 10 seconds.

Thank you for any assistance!

Kristian

**Comment by wills106** [comment]
I have moved this to the discussions section as it isn't an issue with the Integration.

**Comment by 001Tycoon** [comment]
Hello, I have an SPH4000 with USR-W610 Data Transfer Mode is set to Transparent Mode.
Baud rate adaptive "Disable"
I receive all data.
